### DISCUSSION TITLE
Error: Only Deployments created via Git can be promoted to production.

### DISCUSSION THREAD

**Discussion by dasblitz** [vercel/community#866] [Help] [answered]
Since this afternoon we get this error when trying to redeploy deployments (or promote to production) on Vercel:

"Error:Only Deployments created via Git can be promoted to production."

Except, all our deployments are deployed through a github action and nothing in the github actions changed between this afternoon and this morning.

I tried sending a request for support but when sending submit it logs a 500 server error in the console 🤷🏻, so I'll try and ask here.

**Comment by amyegan** [comment]
Hi @dasblitz! Very sorry about the error on the support page. I'm assuming it was on the form at [vercel.com/help](https://vercel.com/help#issues). If you have any additional information about what happened, please let me know. I would like to share it with the team so we can prevent it from happening again.

I want to make sure I understand your original problem correctly. Are you seeing something like this when trying to promote a prebuilt deployment that came from a GitHub Action?

<img width="519" alt="image" src="https://user-images.githubusercontent.com/8247105/190439694-e0d3b83d-85e4-4bb4-9dfe-8a2c1832b24b.png">

If that is the case, then you'll need to use a GitHub Action or manual deployment with the `--prod` flag to rebuild and deploy to production with the correct environment variables and settings.

**Comment by dasblitz** [comment]
Hi Amy,


It was the form on vercel.com/help indeed. I don’t have much more information about the failing form other then this screenshot of the console:




About the actual problem we’re facing on Vercel: Yes that is the error that we see when trying to redeploy any deployment from the Vercel interface. I don’t fully understand your answer. Up until 2 - 3 hours ago we could push commits to a branch on github. This triggers a GitHub Action that deploys to Vercel. Then in the Vercel dashboard we could see all those deployments and redeploy or promote them to production (current). This is something we heavily use. Since 2 - 3 hours this doesn’t work anymore. We see the deployments created from the GitHub action, but on redeploy or promoting to production we get the error:




I don’t understand this error, since the deployments are already created via Git. We also get the same error for deployments created in the past, that we were able to redeploy before this afternoon without problem. 

Kind regards,

Arjen de Vries

**Reply by amyegan** [reply]
I must be misunderstanding your workflow. I can't recall ever being able to redeploy or promote a Deployment that says "via Vercel CLI" because the necessary files aren't available on the server to be rebuilt with the Production settings. But it isn't something I try very often so I could be mistaken!
<img width="652" alt="image" src="https://user-images.githubusercontent.com/8247105/190461956-51d25ab8-2178-44f6-8b84-848962a5370e.png">

Can you walk me through each step of your workflow? I'm also interested in anything that may have changed recently and when your last successful "promote" or "redeploy" happened.

If you can share a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example), that would be most helpful.

**Reply by dasblitz** [reply]
Mm ok it seems our setup is different from what you're showing. We use https://github.com/amondnet/vercel-action in our github action but our deploys never show as being done with vercel cli:

![5A2EAF57-C68A-4177-9272-7FC9353785A6](https://user-images.githubusercontent.com/6230374/190481650-f1f0a393-8f40-480c-ba3f-504c54f5e702.jpeg)

What's also confusing me is that our builds definitely happen on Vercel. (At least I think so, because I can follow the progress of the build in the Vercel UI). While de docs for the amondnet/vercel-action now say the build is supposed to happen in the github action. Almost seems like I had something working that wasn't really supposed to work for over a year, though I still don't understand why it worked until this morning. We haven't touched our github action setup in any way.

**Reply by dasblitz** [reply]
Btw tnx for your reply 🙏🏻 I'm also in contact with Vercel support via email. Not sure which one to follow up now 😅

**Reply by amyegan** [reply]
If you're working with Vercel support via email, please continue with that! You'll be able to discuss more account and deployment details safely that way. I'll add a note to the support case so they know this discussion is related. Thanks for letting me know!

**Comment by leerob** [comment, marked as answer]
Good news everyone, you can now redeploy or promote CLI deployments in the dashboard!

https://vercel.com/changelog/redeploy-or-promote-cli-deployments-from-the-dashboard
